frontend/src/components/accountPage/UserListLogic/UserListLogic.tsx

```diff
@@ -0,0 +1,220 @@
+import { FC, useState, useEffect, useRef, useContext } from 'react';
+import { Spin } from 'antd';
+import {
+  useTenantsQuery,
+  useApplyTenantMutation,
+} from '../../../generated-types';
+import { getTenantPatchJson } from '../../../graphql-components/utils';
+import UserList from '../UserList/UserList';
+import { UserAccountPage } from '../../../utils';
+import { AuthContext } from '../../../contexts/AuthContext';
+import { Role } from '../../../generated-types';
+import { ErrorContext } from '../../../errorHandling/ErrorContext';
+import { ErrorTypes, SupportedError } from '../../../errorHandling/utils';
+
+export interface IUserListLogicProps {
+  workspaceName: string;
+  workspaceNamespace: string;
+}
+const UserListLogic: FC<IUserListLogicProps> = props => {
+  const { apolloErrorCatcher, makeErrorCatcher } = useContext(ErrorContext);
+  const genericErrorCatcher = makeErrorCatcher(ErrorTypes.GenericError);
+
+  const { userId } = useContext(AuthContext);
+  const { workspaceName, workspaceNamespace } = props;
+  const [loadingSpinner, setLoadingSpinner] = useState(false);
+  const [errors, setErrors] = useState<any[]>([]);
+  // Used to handle stop while uploading users from CSV
+  const [abortUploading, setAbortUploading] = useState<boolean>(false);
+  const abortUploadingRef = useRef(false);
+  const [uploadedNumber, setUploadedNumber] = useState<number>(0);
+  const [uploadedUserNumber, setUploadedUserNumber] = useState(0);
+  const [users, setUsers] = useState<UserAccountPage[]>([]);
+  const { data, loading, error } = useTenantsQuery({
+    variables: {
+      labels: `crownlabs.polito.it/${workspaceNamespace}`,
+      retrieveWorkspaces: true,
+    },
+    onError: apolloErrorCatcher,
+    notifyOnNetworkStatusChange: true,
+  });
+
+  useEffect(() => {
+    abortUploadingRef.current = abortUploading;
+  }, [abortUploading]);
+
+  const getManager = () => {
+    return `${workspaceName}-${
+      userId || Math.random().toFixed(10).replace('0.', '')
+    }`;
+  };
```

Comment:
```suggestion
  const getManager = () => `${workspaceName}-${
    userId || makeRandomDigits(10)
  }`;
```
We could put this inside utils maybe